From: Anshuman Khandual <anshuman.khandual@arm.com>
Subject: mm/execmem: make ARCH_WANTS_EXECMEM_LATE depend on EXECMEM
Date: Mon, 9 Dec 2024 08:42:51 +0530

Late initialisation of EXECMEM inherently requires EXECMEM itself, and
hence this dependency should not be left with the subscribing platforms.

Link: https://lkml.kernel.org/r/20241209031251.515903-1-anshuman.khandual@arm.com
Signed-off-by: Anshuman Khandual <anshuman.khandual@arm.com>
Cc: Catalin Marinas <catalin.marinas@arm.com>
Cc: Will Deacon <will@kernel.org>
Cc: Kees Cook <kees@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 arch/Kconfig       |    1 +
 arch/arm64/Kconfig |    2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

--- a/arch/arm64/Kconfig~mm-execmem-make-arch_wants_execmem_late-depend-on-execmem
+++ a/arch/arm64/Kconfig
@@ -113,7 +113,7 @@ config ARM64
 	select ARCH_WANT_FRAME_POINTERS
 	select ARCH_WANT_HUGE_PMD_SHARE if ARM64_4K_PAGES || (ARM64_16K_PAGES && !ARM64_VA_BITS_36)
 	select ARCH_WANT_LD_ORPHAN_WARN
-	select ARCH_WANTS_EXECMEM_LATE if EXECMEM
+	select ARCH_WANTS_EXECMEM_LATE
 	select ARCH_WANTS_NO_INSTR
 	select ARCH_WANTS_THP_SWAP if ARM64_4K_PAGES
 	select ARCH_HAS_UBSAN
--- a/arch/Kconfig~mm-execmem-make-arch_wants_execmem_late-depend-on-execmem
+++ a/arch/Kconfig
@@ -1058,6 +1058,7 @@ config ARCH_WANTS_MODULES_DATA_IN_VMALLO
 
 config ARCH_WANTS_EXECMEM_LATE
 	bool
+	depends on EXECMEM
 	help
 	  For architectures that do not allocate executable memory early on
 	  boot, but rather require its initialization late when there is
_
